.github/workflows/publish-crates.yml

```diff
@@ -0,0 +1,80 @@
+name: Publish Crates
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish (without v prefix)'
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  publish-crates:
+    name: Publish crates to crates.io
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # Update versions in Cargo.toml files
+      - name: Update crate versions
+        run: |
+          VERSION=${{ inputs.version }}
+          # Update version in each Cargo.toml
+          for crate in common crypto serde class-hash; do
+            # Update the package version
+            sed -i "s/^version = \".*\"/version = \"$VERSION\"/" "crates/$crate/Cargo.toml"
+            # Also update any internal dependencies to use the same version
+            sed -i "s/{ path = \"..\/common\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+            sed -i "s/{ path = \"..\/crypto\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+            sed -i "s/{ path = \"..\/serde\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+          done
+
+      - uses: dtolnay/rust-toolchain@stable
+
+      # Verify all crates build and test before publishing
+      - name: Verify crates
+        run: |
+          cargo test -p pathfinder-crypto
+          cargo test -p pathfinder-common
+          cargo test -p pathfinder-serde
+          cargo test -p pathfinder-class-hash
+
+      - name: Login to crates.io
+        run: cargo login ${{ secrets.CRATES_IO_TOKEN }}
+
+      # Publish crates in dependency order with retries
+      - name: Publish crates
+        run: |
+          publish_with_retry() {
+            local package=$1
+            local max_attempts=3
+            local attempt=1
+            
+            while [ $attempt -le $max_attempts ]; do
+              echo "Attempting to publish $package (attempt $attempt/$max_attempts)"
+              if cargo publish -p $package; then
+                return 0
+              fi
+              attempt=$((attempt + 1))
+              [ $attempt -le $max_attempts ] && sleep 30
+            done
+            return 1
+          }
+
+          # First publish crypto as it has no internal dependencies
+          publish_with_retry pathfinder-crypto
+          sleep 30
+          
+          # Publish common which depends on crypto
+          publish_with_retry pathfinder-common
+          sleep 30
+          
+          # Publish serde which depends on common and crypto
+          publish_with_retry pathfinder-serde
+          sleep 30
+          
+          # Finally publish class-hash which depends on common, crypto and serde
+          publish_with_retry pathfinder-class-hash
```

Comment:
Can't we just publish all packages with a single `cargo publish` command?

Reply:
Tracking this in a new issue: #2509